infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/IngressNamingStrategy.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.external;
+
+import io.fabric8.kubernetes.api.model.ServicePort;
+import org.eclipse.che.commons.annotation.Nullable;
+
+public interface IngressNamingStrategy {
```

Comment:
IMHO `IngressNamingStrategy` makes an impression that the useful functionality implementation should do - provide name for ingress. But seems ingress name does not really matter and what does - httpRule for ingress
https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/ExternalServerIngressBuilder.java#L109-L117
Please consider renaming it to `IngressRuleStrategy`, maybe even with one method that returns `IngressRule`

Reply:
I've renamed this to `IngressServiceExposureStrategy`